evacuation: add option to remove PVCs on evacuation (#870)
piraeusdatastore/piraeus-operator: docs/explanation/deletion-policies.md

In order to keep potential fail-over times short, Piraeus Datastore performs the following steps during evacuation:

1. Piraeus Datastore searches for PersistentVolumes (PVs) that:
    *   Are attached on the Node to be evacuated and marks them with `wait-for-reattach.evacuation.piraeus.io/<node-name>`
        annotation. It will later use this annotation to wait for the PV to be reattached.
    *   If the PV has a `linstor.csi.linbit.com/evacuation-action` annotation or the storage class has the
        `property.linstor.csi.linbit.com/Aux/linstor.csi.linbit.com/evacuation-action` parameter set, the value is used
        to determine the evacuation action. Possible actions are:
        * `AffinityOverride`: Annotate the PV with `override.piraeus.io/<nodename>`. This causes the LINSTOR Affinity
          Controller to temporarily mark the PV as accessible from any Node.
        * `Delete`: Delete the PVC and PV, causing the **existing data** to be **lost**. This is _only_ useful for PVCs

>           storing caches or otherwise ephemeral data that get recreated automatically, using a StatefulSets
>           `volumeClaimTemplates` or similar mechanisms.

Reviewer: ```suggestion
          storing caches or otherwise ephemeral data that get recreated automatically, for example using a StatefulSet
          with `volumeClaimTemplates`.
```

Or if you want to keep it as it was, I think it should be `StatefulSet's`.

1. Piraeus Datastore searches for PersistentVolumes (PVs) that:
    * Are attached on the Node to be evacuated and marks them with `wait-for-reattach.evacuation.piraeus.io/<node-name>`
      annotation. It will later use this annotation to wait for the PV to be reattached.
    * If the PV has a `linstor.csi.linbit.com/evacuation-action` annotation or the storage class has the
      `property.linstor.csi.linbit.com/Aux/linstor.csi.linbit.com/evacuation-action` parameter set, the value is used
      to determine the evacuation action. Possible actions are:
        * `AffinityOverride`: Annotate the PV with `override.piraeus.io/<nodename>`. This causes the LINSTOR Affinity
          Controller to temporarily mark the PV as accessible from any Node.

>         * `Delete`: Delete the PVC and PV, assuming that they get recreated on moving the workload to another node. This

Reviewer: For my understanding: What component would re-create the PVC in this case? In basic usage, PVCs are manually created by applying a yaml, so in that case they won't automatically come back.

Author: Expanded a bit on this, mainly StatefulSets and some other Operators would be the ones creating the PVCs

          causes the existing data to be lost.
        * `None`: No special action is taken to ensure the PV can be evacuated.

>       If no action is specified, PVs using the `allowRemoteVolumeAccess: "false"` policy, or PVs that are otherwise

Reviewer: In the GitHub preview this appears as part of the last bullet point. Perhaps we need to indent the sub-list more?

Author: Fixed. Checked with `mkdocs serve` that the output looks good.
